test/ng/directive/ngRepeatSpec.js

```diff
@@ -400,6 +400,44 @@ describe('ngRepeat', function() {
       expect(element.find('input')[1].checked).toBe(true);
       expect(element.find('input')[2].checked).toBe(true);
     }));
+
+    //https://github.com/angular/angular.js/issues/16776
+    it('should invoke track by with correct locals', function() {
+      scope.trackBy = jasmine.createSpy().and.callFake(function(k, v) {
+        return [k, v].join('');
+      });
+
+      element = $compile(
+        '<ul>' +
+          '<li ng-repeat="(k, v) in [1, 2] track by trackBy(k, v)"></li>' +
+        '</ul>')(scope);
+      scope.$digest();
+
+      expect(scope.trackBy).toHaveBeenCalledTimes(2);
+      expect(scope.trackBy.calls.argsFor(0)).toEqual([0, 1]);
+      expect(scope.trackBy.calls.argsFor(1)).toEqual([1, 2]);
+    });
```

Comment:
Does this test fail without the fix, or are you just adding one extra test (for an already working behavior)?

Reply:
This one does not fail, just adding an extra one. I'll remove the comment referencing the bug on this one.